bundle/config/validate/validate_sync_patterns.go

```diff
@@ -49,7 +50,8 @@ func checkPatterns(patterns []string, path string, rb bundle.ReadOnlyBundle) (di
 
 	for i, pattern := range patterns {
 		index := i
-		p := pattern
+		// If the pattern starts with !, remove it for the check below
+		p := strings.TrimPrefix(pattern, "!")
```

Comment:
this is done to make sure we don't raise a warning on negation pattern having no matches

Reply:
Could you modify the comment to say why we remove the `!`? That we want to warn users if a glob pattern with negation matches all files in their bundle.